### PR TITLE
Example job update to fix buildNeedsWorkspace to retain backward compatibility after updating to job-dsl 1.42

### PR DESCRIPTION
**Pull request by marcesher**
Update example build flow job to use a configure block for buildNeedsWorkspace that works with job-dsl 1.42, which now includes buildNeedsWorkspace; this new block is compatible with all versions of job-dsl, not just pre-1.42

### CODE REVIEW

File: example-jobs/SampleBuildFlow-example.groovy [hunk 1]
```diff
@@ -44,7 +44,9 @@ customFlowJobWithWorkspace.with{
         git('https://github.com/cfpb/qu.git', 'master')
     }
 
+    //for use on job-dsl-plugin versions 1.41 and earlier
+    //on 1.42+, simply use buildNeedsWorkspace(). See https://jenkinsci.github.io/job-dsl-plugin/#method/javaposse.jobdsl.dsl.jobs.BuildFlowJob.buildNeedsWorkspace
```
Comment (Scotchester) [inline example-jobs/SampleBuildFlow-example.groovy]: Could you modify the formatting of this comment to clarify whether or not the period at the end of `buildNeedsWorkspace().` is necessary, or just separating clauses in the comment?

Comment (imuchnik) [inline example-jobs/SampleBuildFlow-example.groovy]: separating comments

Comment (marcesher) [inline example-jobs/SampleBuildFlow-example.groovy]: @Scotchester will do

**Comment by Scotchester** [comment]
:+1: Thanks!

**Pull request merged**